### PR TITLE
i18n - sasl  internationalization + refactoring

### PR DESCRIPTION
**Pull request by honza889**
i18n internationalization of `sasl` package
* SaslExceptions are now prefixed with `[MECHANISM-NAME]`

refactoring:
* Digest
 * variable `md5` renamed to `digest`
 * in `createCipher()` merged two cipher switches
 * in `_private.DigestUtil` removed unused methods params (`len`)
* `ScramUtil.getTwoWayPasswordChars()` method accept `TwoWayPassword` only

Note: `log.nullParameter(String parameter)` message accept parameter name as String - I hope translating names of parameters of methods is not required.

### CODE REVIEW

**Comment by honza889** [comment]
Will need rebase after #180 will be merged.

File: src/main/java/org/wildfly/security/sasl/anonymous/AnonymousSaslClient.java [hunk 1]
```diff
@@ -86,6 +87,6 @@ protected AnonymousSaslClient(final String protocol, final String serverName, fi
                 negotiationComplete();
                 return b.toArray();
         }
-        throw new SaslException("Invalid state");
+        throw new IllegalStateException();
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/sasl/anonymous/AnonymousSaslClient.java]: Now we can use ``Assert.unreachableCode`` and/or ``Assert.impossibleSwitchCase`` for this case.

**Comment by dmlloyd** [comment]
Looks OK to me other than my previous minor note.

**Comment by honza889** [comment]
Fixed to use `Assert.unreachableCode` and `Assert.impossibleSwitchCase` by dmlloyd note.

**Comment by honza889** [comment]
Refactored and prepared for merge.

**Pull request merged**